p2p/src/authenticated/connection/handshake.rs

```diff
@@ -4,14 +4,20 @@ use bytes::Bytes;
 use commonware_cryptography::{PublicKey, Scheme};
 use commonware_macros::select;
 use commonware_runtime::{Clock, Sink, Spawner, Stream};
+use commonware_utils::union;
 use prost::Message;
 use std::time::{Duration, SystemTime, UNIX_EPOCH};
 
-const NAMESPACE: &[u8] = b"_COMMONWARE_P2P_AUTHENTICATED_HANDSHAKE_";
+const NAMESPACE_SUFFIX_HANDSHAKE: &[u8] = b"_HANDSHAKE";
+
+fn suffix_namespace(namespace: &[u8]) -> Vec<u8> {
+    union(namespace, NAMESPACE_SUFFIX_HANDSHAKE)
+}
 
 pub fn create_handshake<E: Clock, C: Scheme>(
     runtime: E,
     crypto: &mut C,
+    namespace: &[u8],
```

Comment:
I wonder if there is a way to provide the already suffixed namespace here to avoid unnecessary allocations?

Reply:
Created and assigned myself the issue [here](https://github.com/commonwarexyz/monorepo/issues/181)